source/dem/find_contact_detection_step.cc

```diff
@@ -4,31 +4,26 @@ using namespace dealii;
 
 template <int dim>
 bool
-find_contact_detection_step(
-  Particles::ParticleHandler<dim> &particle_handler,
-  const double &                   dt,
-  const double &                   smallest_contact_search_criterion,
-  MPI_Comm &                       mpi_communicator,
-  bool &                           sorting_in_subdomains_step,
-  std::unordered_map<types::particle_index, double> &displacement)
+find_contact_detection_step(Particles::ParticleHandler<dim> &particle_handler,
+                            const double &                   dt,
+                            const double &smallest_contact_search_criterion,
+                            MPI_Comm &    mpi_communicator,
+                            bool &        sorting_in_subdomains_step,
+                            std::vector<double> &displacement)
 {
-  double       max_displacement       = 0;
-  unsigned int contact_detection_step = 0;
-
-  // Looping through all the particles:
   if (sorting_in_subdomains_step)
-    {
-      // If last step was a sorting into subdomains step, the displcement
-      // is reinitialized and then the new displacement is calculated
+    for (auto &d : displacement)
+      d = 0.;
 
-      // Clearing displacement
-      displacement.clear();
-    }
+  double       max_displacement       = 0;
+  unsigned int contact_detection_step = 0;
 
   // Updating displacement
   for (auto &particle : particle_handler)
     {
-      auto &particle_properties   = particle.get_properties();
+      auto &particle_properties = particle.get_properties();
+      if (displacement.size() <= particle.get_id())
+        displacement.resize(particle.get_id() + 1);
```

Comment:
Again, can't that pause a problem when id is large? Maybe there is something I don't understand, but isnt there an issue with sizing vectors with particle id?